### DISCUSSION TITLE
StageResult - How to retrieve this object to get all the steps information

### DISCUSSION THREAD

**Discussion by sridharreddysuram123** [allure-framework/allure2#2119] [Bug Reports] [answered]
This will help us in logging bugs with all the steps in E2E integration. 
How to retrieve all the test steps(step names) executed in one test case. I want to catch this in AfterMethod or onTestFailure in TestNG. 
I saw that StageResult has the steps info, but how to get StageResult of the currentTest in Allure? 

Please help me. Thank you.

**Comment by baev** [comment, marked as answer]
You can use Allure Lifecycle API to access the current test context:

```java
Allure.getLifecycle().updateTestResult(tr -> ...);
```
